api/src/main/java/io/opentelemetry/api/trace/propagation/HttpTraceContext.java

```diff
@@ -5,78 +5,17 @@
 
 package io.opentelemetry.api.trace.propagation;
 
-import static io.opentelemetry.api.internal.Utils.checkArgument;
-
-import io.opentelemetry.api.internal.TemporaryBuffers;
-import io.opentelemetry.api.trace.Span;
-import io.opentelemetry.api.trace.SpanContext;
-import io.opentelemetry.api.trace.SpanId;
-import io.opentelemetry.api.trace.TraceFlags;
-import io.opentelemetry.api.trace.TraceId;
-import io.opentelemetry.api.trace.TraceState;
-import io.opentelemetry.api.trace.TraceStateBuilder;
 import io.opentelemetry.context.Context;
 import io.opentelemetry.context.propagation.TextMapPropagator;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Objects;
-import java.util.Set;
-import java.util.logging.Logger;
-import java.util.regex.Pattern;
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
 
-/**
- * Implementation of the TraceContext propagation protocol. See <a
- * href=https://github.com/w3c/distributed-tracing>w3c/distributed-tracing</a>.
- */
 @Immutable
+@Deprecated
```

Comment:
If you could add some javadoc with a pointer over to what the user should be using, that would be more user-friendly. Also on the `getInstance()` method, I think.